ppr-ui/tests/unit/UnitNotePanels.spec.ts

```diff
@@ -124,63 +220,44 @@ describe('UnitNotePanels', () => {
-        const effectiveDate = content.findAll('h3').at(0).text()
-        const effectiveDateTime = content.findAll('.info-text.fs-14').at(0).text()
-        expect(effectiveDate).toBe('Effective Date and Time')
-        expect(effectiveDateTime).toBe(pacificDate(item.effectiveDateTime))
-      }
-
-      // Check the expiry date and time
-      if (item.expiryDateTime) {
-        const expiryDate = content.findAll('h3').at(1).text()
-        const expiryDateTime = content.findAll('.info-text.fs-14').at(1).text()
-        expect(expiryDate).toBe('Expiry Date and Time')
-        expect(expiryDateTime).toBe(pacificDate(item.expiryDateTime))
-      }
-
-      // Check the remarks
-      if (item.remarks) {
-        const remarks = content.findAll('h3').at(2).text()
-        const remarksText = content.findAll('.info-text.fs-14').at(2).text()
-        expect(remarks).toBe('Remarks')
-        expect(remarksText).toBe(item.remarks)
-      }
-
-      // Check the person giving notice table
-      const noticeTable = content.find('#persons-giving-notice-table')
-      expect(noticeTable.exists()).toBe(true)
-
-      // Check the notice party data
-      const noticeParty = item.givingNoticeParty
-      if (noticeParty) {
-        const noticePartyIcon = noticeTable.findComponent({ name: 'VIcon' })
-        expect(noticePartyIcon.exists()).toBe(true)
-
-        const mdiIconClass = Array.from(noticePartyIcon.element.classList)
-          .find(className => className.startsWith('mdi-'))
-        expect(mdiIconClass).toBe(wrapper.vm.getNoticePartyIcon(noticeParty))
-
-        const noticePartyName = noticeTable.find('.notice-party-name')
-        expect(noticePartyName.exists()).toBe(true)
-        expect(noticePartyName.text()).toContain(wrapper.vm.getNoticePartyName(noticeParty))
-
-        // Check the notice party address, email, and phone number
-        const noticePartyAddress = noticeTable.findComponent(BaseAddress)
-        expect(noticePartyAddress.exists()).toBe(true)
-        expect(noticePartyAddress.props('value')).toBe(noticeParty.address)
-
-        const noticePartyEmail = noticeTable.find('td:nth-child(3)')
-        expect(noticePartyEmail.exists()).toBe(true)
-        expect(noticePartyEmail.text()).toBe(noticeParty.emailAddress)
-
-        const noticePartyPhone = noticeTable.find('td:nth-child(4)')
-        expect(noticePartyPhone.exists()).toBe(true)
-        expect(noticePartyPhone.text()).toBe(noticeParty.phoneNumber)
+      verifyBodyContent(note, content)
+
+      panelIdx++
+      noteIdx++
+
+      // Check the data for additionally grouped notes
+      if ([UnitNoteDocTypes.EXTENSION_TO_NOTICE_OF_CAUTION, UnitNoteDocTypes.CONTINUED_NOTE_OF_CAUTION]
+        .includes(note.documentType)) {
+        // Verify additional grouped notes ending with the notice of caution
+        let i = noteIdx
+        let additionalNotePos = 1
+        while (mockUnitNotes[i].documentType !== UnitNoteDocTypes.NOTICE_OF_CAUTION) {
```

Comment:
Unit Tests should be simple 😆

Reply:
Sadly the logic is not simple for the panel, but I wanted to ensure there can be a sense of assurance of it working properly from a test. If you can think of any revisions to simplify it please let me know. I am aware that this might be a bit more difficult to maintain than the other tests, but I don't think its overly complicated at this point.